### DISCUSSION TITLE
Backpack Basset failed to check it's working properly.

### DISCUSSION THREAD

**Discussion by r0aringthunder** [Laravel-Backpack/community-forum#1136] [Q&A (Help)] [answered]
Hey artisans!

I'm hoping someone can give me some great guidance here. I'm trying to upgrade to Backpack V6. Below I am adding my composer.json, filesystems.php and what I've done so far.

# .env
```env
APP_ENV=development
# Tried with both `development` and `local`
```

# composer.json
```json
    "repositories": {
        "backpack": {
            "type": "composer",
            "url": "https://repo.backpackforlaravel.com/"
        },
    },
    "require": {
        ...
        "backpack/crud": "^6.0",
        "backpack/editable-columns": "^3.0",
        "backpack/pro": "^2.0",
        "backpack/theme-coreuiv2": "^1.2",
        ...
    },
    "require-dev": {
        "backpack/devtools": "^3.1",
        "backpack/generators": "^4.0",
        ...
    }
```
# filesystems.php
```php
<?php

return [
    'default' => env('FILESYSTEM_DISK', 'local'),

    'disks' => [
        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
            'throw' => false,
        ],
    ],

    'links' => [
        public_path('storage') => storage_path('app/public'),
    ],

];
```

I have run the following commands with their responses
```bash
$> php artisan storage:link
 INFO  The [public/storage] link has been connected to [storage/app/public].
 
 
 $> php artisan backpack:install (Selected coreuiv2 and added all the packages like devtools after that prompt)
   STEP 1  Installing Backpack CRUD.

  Publishing configs, views, routes and assets ..................................................................... DONE
  Creating users table ............................................................................................. DONE
  Creating CheckIfAdmin middleware ................................................................................. DONE
  Installing Generators ............................................................................................ DONE
  Installing Basset ................................................................................................ DONE

   STEP 2  Installing a Theme.

  │ Choose your admin UI, depending on your project and preferences.

  CoreUIv2 was already installed



   STEP 3  Creating an admin.

  │ Quickly jump in your admin panel, using the email & password you choose here.
  │ Currently there are 21174 users in the users table.
  │ Skipping creating an admin user.

   INFO  Checking Backpack Basset installation.  

  Initializing basset check ........................................................................................ DONE  
  Checking cache storage ........................................................................................... DONE  
  Fetching a basset ................................................................................................ ERROR  

   ERROR  Error fetching the file.  

  │ Backpack Basset failed to check it\'s working properly.
  │
  │ This may be due to multiple issues. Please ensure:
  │  1) APP_URL is correctly set in the .env file.
  │  2) Your server is running and accessible at https://development.domain.net.
  │  3) The public disk is properly configured in config/filesystems.php.
  │     Optionally, basset provides a disk named "basset", you can use it instead.
  │  4) The storage symlink exists and is valid (by default: public/storage).
  │
  │ For more information and solutions, please visit the Backpack Basset FAQ at:
  │ https://github.com/laravel-backpack/basset#faq
  
  
$> php artisan storage:unlink
```
![Screenshot 2024-08-30 181148](https://github.com/user-attachments/assets/3710c363-8c66-4448-a7f2-1ae3c17bcb8f)

Then tried to rinse and repeat within my docker container as well as on the host machine to see if I got any other response but I keep coming to this error

**Comment by r0aringthunder** [comment]
If someone has ran into this issue before it would help a ton to get more insight 🙂

**Reply by pxpm** [reply]
Hey @r0aringthunder also worth checking if you added the `volume` in your docker config ? 
We have demo setup with docker: https://github.com/Laravel-Backpack/demo/blob/main/docker-compose.yml

Cheers

**Comment by jcastroa87** [comment]
Hello @r0aringthunder 

Did you check end .env the APP_URL is correctly config? Need to be the real URL of app.
Then check the permissions:
```
sudo chmod -R 775 storage
sudo chmod -R 775 bootstrap/cache
```

You can try to remove and create again storage link:
```
php artisan storage:unlink
```

Let me know if thats help for you.

Cheers.

**Reply by r0aringthunder** [reply, marked as answer]
I'll have to loop back to this ticket for upgrading backpack soon but thank you for the help! I'll let you know if this fixed my issue or not in a couple days if not later today 😄

**Comment by jcastroa87** [comment]
Due to not activity, I will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.

**Comment by razlupercio** [comment]
I'm having this exact issue with laravel 12 and backpack 6 on Windows, tried the suggestions on https://github.com/Laravel-Backpack/basset/blob/main/readme.md#faq without success.

**Comment by wfgm5k2d** [comment]
Проблема сохраняется на laravel 10, backpack 6
"backpack/backupmanager": "^5.0",
"backpack/crud": "^6.8.4",
"backpack/filemanager": "^3.0",
"backpack/logmanager": "^5.0",
"backpack/pro": "2.0",
"backpack/theme-coreuiv2": "^1.2",
"laravel/framework": "^10.48.25",

root@bb529850d940:/var/www# php artisan backpack:install

   STEP 1  Installing Backpack CRUD.

  Publishing configs, views, routes and assets ........................................................................... DONE
  Creating users table ................................................................................................... DONE
  Creating CheckIfAdmin middleware ....................................................................................... DONE
  Installing Generators .................................................................................................. DONE
  Installing Basset ...................................................................................................... DONE

   STEP 2  Installing a Theme.

  │ Choose your admin UI, depending on your project and preferences.

  CoreUIv2 was already installed



   STEP 3  Creating an admin.

  │ Quickly jump in your admin panel, using the email & password you choose here.
  │ Currently there are 1432 users in the users table.
  │ Skipping creating an admin user.


   STEP 4  Installing premium Backpack add-ons:

  │ Add tons of features and functionality to your admin panel, using our paid add-ons.
  │ For more information, payment and access please visit https://backpackforlaravel.com/pricing

  1 Backpack Pro .................................................................................................... INSTALLED
    Adds 5 operations, 10 filters, 28 fields, 6 columns, charts

  2 DevTools .................................................................................................... NOT INSTALLED
    Helps generate models, migrations, operations and CRUDs

  3 Editable Columns ............................................................................................ NOT INSTALLED
    Allow your admins to make small changes from the table view



   INFO  Checking Backpack Basset installation.  

  Initializing basset check ................................................................................................................... DONE  
  Checking cache storage ...................................................................................................................... DONE  
  Fetching a basset .......................................................................................................................... ERROR  

   ERROR  cURL error 7: Failed to connect to localhost port 8000 after 0 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://localhost:8000/storage/basset/vendor/backpack/basset/tests/Helpers/basset-test.js.  

  │ Backpack Basset failed to check it's working properly.
  │
  │ This may be due to multiple issues. Please ensure:
  │  1) APP_URL is correctly set in the .env file.
  │  2) Your server is running and accessible at http://localhost:8000.
  │  3) The public disk is properly configured in config/filesystems.php.
  │     Optionally, basset provides a disk named "basset", you can use it instead.
  │  4) The storage symlink exists and is valid (by default: public/storage).
  │
  │ For more information and solutions, please visit the Backpack Basset FAQ at:
  │ https://github.com/laravel-backpack/basset#faq
